haystack/nodes/prompt/prompt_node.py

```diff
@@ -527,15 +532,58 @@ def invoke(self, *args, **kwargs):
         return responses
 
     @classmethod
-    def supports(cls, model_name_or_path: str) -> bool:
-        return any(m for m in ["ada", "babbage", "davinci", "curie"] if m in model_name_or_path)
+    def supports(cls, model_name_or_path: str, **kwargs) -> bool:
+        valid_model = any(m for m in ["ada", "babbage", "davinci", "curie"] if m in model_name_or_path)
+        return valid_model and kwargs.get("base_url") is None
```

Comment:
I think relying on whether the `base_url` is provided to distinguish between the Azure API and OpenAI API feels a bit fragile. I think I'd already feel more confident about this if we renamed this variable to `azure_base_url`. What do you think @julian-risch ?

Reply:
Let's rely on both azure parameters as they are both mandatory for Azure

Reply:
Relying on both parameters makes sense to me and in addition I would rename them to `azure_base_url` and `azure_deployment_name`. Makes it easier to understand in the init method when to set them.

Reply:
I updated both in the prompt node and openai answer generator. I also updated `test_generator.py` tests to support both OpenAI and Azure.